specs/test_formats/README.md

```diff
@@ -0,0 +1,170 @@
+The way this list of forks is interpreted, is up to the test-runner:
+State-transition test suites may want to just declare forks that are being covered in the test suite,
+ whereas shuffling test suites may want to declare a list of forks to test the shuffling algorithm for individually. 
+
+### Test completeness
+
+Tests should be independent of any sync-data. If one wants to run a test, the input data should be available from the YAML.
+The aim is to provide clients with a well-defined scope of work to run a particular set of test-suites.
+
+- Clients that are complete are expected to contribute to testing, seeking for better resources to get conformance with the spec, and other clients.
+- Clients that are not complete in functionality can choose to ignore suites that use certain test-runners, or specific handlers of these test-runners.
+- Clients that are on older versions can test there work based on older releases of the generated tests, and catch up with newer releases when possible.
+
+## Test Suite
+
+```
+title: <required, string, short, one line> -- Display name for the test suite
+summary: <required, string, average, 1-3 lines> -- Summarizes the test suite
+forks_timeline: <required, string, reference to a fork definition file, without extension> -- Used to determine the forking timeline
+forks: <required, list of strings> -- Runner decides what to do: run for each fork, or run for all at once, each fork transition, etc.
+  - ... <required, string, first the phase name, then the spec version>
+config: <required, string, reference to a config file, without extension> -- Used to determine which set of constants to run (possibly compile time) with
+runner: <required, string, no spaces, python-like naming format> *MUST be consistent with folder structure*
+handler: <optional, string, no spaces, python-like naming format> *MUST be consistent with folder structure*
+
+test_cases: <list, values being maps defining a test case each>
+   ...
+
+```
+
+## Config
+
+A configuration is a separate YAML file.
+Separation of configuration and tests aims to:
+- prevent duplication of a minimal set of tests
+- make all tests easy to upgrade when a new config constant is introduced.
+- clearly define which constants to use
+- share-able between clients, for cross-client short or long lived test-nets
+- minimize the amounts of different constants permutations to compile as a client. \**
+ 
+\**: Some clients prefer compile-time constants and optimizations.
+They should compile for each configuration once, and run the corresponding tests per build target.
+
+## Fork-timeline
+
+A fork timeline is (preferably) loaded in as a configuration object into a client, as opposed to the constants configuration:
+ - we do not allocate or optimize any code based on slot numbers
+ - when we transition from one fork to the other, it is preferred to stay online.
+ - we may decide on a slot number for a fork based on external events (e.g. Eth1 log event),
```

Comment:
Eth1 log event defines the genesis time, not the slot number.

Reply:
Yes, just want to highlight that it can be a dynamically chosen slot, and doesn't have to be a pre-determined thing when compiling.